device-driver/examples/test-macro-driver.rs

```diff
@@ -249,11 +287,11 @@ fn main() {
     test_device.foo().write(|w| w).unwrap();
     assert_eq!(test_device.foo().read().unwrap().value(), 0x1234);
 
-    test_device.bar().clear().unwrap();
+    test_device.cfg_reg().clear().unwrap();
     #[cfg(windows)]
-    assert_eq!(test_device.bar().read().unwrap().value(), 0x1234);
+    assert_eq!(test_device.cfg_reg().read().unwrap().value(), 0x1234);
     #[cfg(not(windows))]
-    assert_eq!(test_device.bar().read().unwrap().value(), 0x5678);
+    assert_eq!(test_device.cfg_reg().read().unwrap().value(), 0x5678);
 
     test_device.sleep().dispatch().unwrap();
     assert_eq!(test_device.last_command, 0);
```

Comment:
Can we get some testing of the blocks here? This example has the most usage of all apis, so it'd be nice to add this new api here as well